migrator/migrator.go

```diff
@@ -449,9 +450,12 @@ func (m Migrator) MigrateColumn(value interface{}, field *schema.Field, columnTy
 
 	// check default value
 	if v, ok := columnType.DefaultValue(); ok && v != field.DefaultValue {
-		// not primary key
-		if !field.PrimaryKey {
-			alterColumn = true
+		// not all equal null
+		if !(utils.CheckColumnDefaultNull(v) && utils.CheckColumnDefaultNull(field.DefaultValue)) {
```

Comment:
We can also check `MariaDB` version in mysql driver to replace string "NULL" instead of using `tils.CheckColumnDefaultNull(v)`directly, but I don't think it is necessary.